nixos/modules/services/networking/peroxide.nix

```diff
@@ -0,0 +1,134 @@
+          and place it in the service's configuration directory for the peroxide service
+          to start.
+        '';
+      };
+
+      organization = mkOption {
+        type = types.str;
+        default = "N/A";
+        description = "Value for the 'Organization' field of the self-signed X.509 certificate.";
+      };
+
+      hostname = mkOption {
+        type = types.str;
+        default = config.networking.hostName;
+        defaultText = "nixos";
+        description = "Value for the 'CN' field of the self-signed X.509 certificate.";
+      };
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          UserPortImap = mkOption {
+            type = types.port;
+            default = 1143;
+            description = "The port on which to listen for IMAP connections.";
+          };
+
+          UserPortSmtp = mkOption {
+            type = types.port;
+            default = 1025;
+            description = "The port on which to listen for SMTP connections.";
+          };
+
+          ServerAddress = mkOption {
+            type = types.str;
+            default = "[::0]";
+            example = "localhost";
+            description = "The address on which to listen for connections.";
+          };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for peroxide.  See
+        <link xlink:href="https://github.com/ljanyst/peroxide/blob/master/config.example.yaml"/>
+        for an example configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.peroxide.settings = {
+      # peroxide deletes the cache directory on startup, which requires write
+      # permission on the parent directory, so we can't use
+      # /var/cache/peroxide
+      #
+      # We also need peroxide-cfg to be able to read the cache when run as user
+      # peroxide *outside* the service (i.e. sudo -u peroxide peroxide-cfg ...)
+      # in order to configure the bridge, because this tool also accesses the
+      # peroxide cache.  Using systemd CacheDirectory= with DynamicUser=true
+      # makes this impossible because systemd sets up permissions such that the
+      # directory is unreadable by non-root users outside of the service unit,
+      # but this restriction is not applied to ConfigurationDirectory=, so
+      # putting the peroxide cache in there is our only option if we want users
+      # to be able to run peroxide-cfg unpriveleged.
+      #
+      # (A better upstream implementation would be for the config tool to talk
+      # to the service instead of writing to files directly.)
+      CacheDir = "/etc/peroxide/cache";
+    };
+
+    systemd.services.peroxide = {
+      requires = [ "network.target" ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        DynamicUser = true;
```

Comment:
All that documentation/explanation written above just because of this option. I think that means you shouldn't use this option.

Reply:
I initially did not, but @SuperSandro2000 requested changing to use `DynamicUser` in their review.

Reply:
If it does not easily work, then we can't use it but we definitely don't need to hardcode an ID. That can be dynamically allocated.

Reply:
What is the correct way to dynamically allocate a system user (ID)?

Reply:
`users.users.foo.isSystemUser`